src/OpenTelemetry.AutoInstrumentation/Configuration/EnvironmentConfigurationTracerHelper.cs

```diff
@@ -56,10 +57,14 @@ public static TracerProviderBuilder UseEnvironmentVariables(this TracerProviderB
     public static TracerProviderBuilder AddSdkAspNetInstrumentation(this TracerProviderBuilder builder)
     {
 #if NET462
-        return builder.AddAspNetInstrumentation();
-#elif NETCOREAPP3_1_OR_GREATER
-        return builder.AddAspNetCoreInstrumentation();
+        builder.AddAspNetInstrumentation();
 #endif
+        if (Assembly.GetEntryAssembly().GetReferencedAssemblies().Any(x => x.Name == "Microsoft.AspNetCore"))
```

Comment:
Notice: We cannot detect `Microsoft.AspNetCore.Http` dll directly as it's probably a subdependency of a dependency. I think it's reasonable to assume, if `Microsoft.AspNetCore` is present then `Microsoft.AspNetCore.Http` is also present.

Reply:
Another important one: `Assembly.GetEntryAssembly()` returns null if application is created in unmanaged process eg IIS. Wondering if the same happens when ASP.NET Core is inproc hosted in IIS.

Reply:
One way is to cut off the feature: run ASP.NET Core with .NET Framework.

Reply:
To identify ASP.NET, we could use `Assembly.GetAssembly()` API and check for `System.Web.dll`. Similar check is needed for ASP.NET MVC/Web API.

Reply:
> To identify ASP.NET, we could use `Assembly.GetAssembly()` API and check for `System.Web.dll`. Similar check is needed for ASP.NET MVC/Web API.

As far as I understand Assembly.GetAssembly() etc requires assembly to be loaded. We need to know before if the assembly exists / will be loaded / is loadable, because the StartupHook runs early.

Reply:
ASP.NET does not use StartupHook, by the time we reach this code all the ASP.NET libraries are loaded and we could use this approach for .NET Framework. 

But, I agree to your comment - https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/932#discussion_r920326457 we may not need to do anything special here.